Update Developer Guide (Draft) (#104)
AY2021S1-CS2113-T13-1/tp: docs/DeveloperGuide.md

This document describes the design, implementation and architecture of NotUS.

#### <a id="setup"><ins>1.1 Setting up and getting started</ins></a>

> Prerequisites:
> <ul>JDK 11</ul>
> <ul>IntelliJ IDE</ul>

Reviewer: Easier to make changes if needed - using markdown syntax
```suggestion
**Prerequisites:**
* JDK 11
* IntelliJ IDE
```

Fork this repo and clone it onto your local machine.

> Import the project as a <b>Gradle project</b>.

Reviewer: ```suggestion
Import the project as a **Gradle project**.
```

  <br><em>Figure 1</em>
</p>

The architecture design is given in the diagram above. The main components of NotUS are:

1. `InterfaceManager`: Manages the user input as well as the message output from application.
1. `Parser`: Makes sense of user input, and parses the information to the respective commands.
1. `Command`: Executes the necessary tasks, depending on the respective command calls .
1. `TagManager`: Stores and manages the creation and deletion of tags and other tag-related functionality.
1. `Timetable`: Stores and manages the creation and deletion of events and other event-related functionality.
1. `Notebook`: Stores and manages the creation and deletion of notes and other note-related functionality.
1. `StorageManager`: Manages the loading of existing saved files and exporting of data to human-editable files

> <b>NotUS</b> manages the flow of the application. On launch, it will create the necessary components, as listed above

Reviewer: ```suggestion
**NotUS** manages the flow of the application. On launch, it will create the necessary components, as listed above
```

 and then attempts to load any existing saved files into the application. Subsequently, it will accept and interpret
the user input and execute the commands accordingly. The diagram below depicts the main flow of the application.

{insert Notus.puml here}

<br>

> <b>Parser</b>

Reviewer: ```suggestion
**Parser**
```

> <li>Receives the user input message as a whole.</li>
> <li>Interprets the type of command and splits the message to identify all the parameters provided.</li>
> <li>Creates and returns the Command class respectively.</li>
>  
>  {insert Parser.puml here}
> 
> <b>Command</b> classes

Reviewer: ```suggestion
1. Receives the user input message as a whole.
1. Interprets the type of command and splits the message to identify all the parameters provided.
1. Creates and returns the Command class respectively.
 
 {insert Parser.puml here}

**Command** classes
```
